File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.cs
using System;

public class Solution 
{
    public long[] solution(long[] numbers) 
    {
        long[] answer = new long[numbers.Length];

        for (int i = 0; i < numbers.Length; i++)
        {
            if (numbers[i] % 2 == 0)
            {
                answer[i] = numbers[i] + 1;
            }

            else
            { 
                long lastZero = (numbers[i] + 1) & (-numbers[i]);
                long zeroToOne = numbers[i] | lastZero;
                long temp = (zeroToOne & (~(lastZero) >> 1));

                answer[i] = temp;
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.cpp
#include <string>
#include <vector>
 
using namespace std;
 
vector<long long> solution(vector<long long> numbers)
{
    vector<long long> answer;
 
    for (int i = 0; i < numbers.size(); i++)
    {
        if (numbers[i] & 1)
        {
            for (int j = 0; j < 64; j++)
                if (!(numbers[i] & ((long long)1 << j)))
                {
                    answer.push_back(numbers[i] + ((long long)1 << (j - 1)));
                    break;
                }
        }
        else answer.push_back(numbers[i] + 1);
    }
 
    return answer;
}